mmcv/runner/ipu/dataloder.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from collections.abc import Mapping, Sequence
+
+import poptorch
+from functools import partial
+from torch.utils.data.dataloader import default_collate
+
+from mmcv.parallel.data_container import DataContainer
+
+
+def collate(batch, samples_per_gpu=1):
+    """Puts each data field into a tensor/DataContainer with outer dimension
+    batch size.
+
+    Extend default_collate to add support for
+    :type:`~mmcv.parallel.DataContainer`. There are 3 cases.
+
+    1. cpu_only = True, e.g., meta data
+    2. cpu_only = False, stack = True, e.g., images tensors
+    3. cpu_only = False, stack = False, e.g., gt bboxes
+    """
+
+    if not isinstance(batch, Sequence):
+        raise TypeError(f'{batch.dtype} is not supported.')
+
+    if isinstance(batch[0], DataContainer):
+        raise TypeError('DataContainer is not supported in ipu data loader.')
+    elif isinstance(batch[0], Sequence):
+        transposed = zip(*batch)
+        _list = []
+        for samples in transposed:
+            if not isinstance(samples[0], DataContainer):
+                _list.append(collate(samples, samples_per_gpu))
+        return _list
+    elif isinstance(batch[0], Mapping):
+        _dic = {}
+        for key in batch[0]:
+            if not isinstance(batch[0][key], DataContainer):
+                _dic[key] = collate([d[key] for d in batch])
+        return _dic
+    else:
+        return default_collate(batch)
+
+
+class IPUDataloader(poptorch.DataLoader):
+    """ Thin wrapper around the traditional `torch.utils.data.DataLoader`
+    to abstract away some of the batch sizes calculationsand speed up
+    data loading.
+
+    If this data loader is used in a distributed execution environment, it will
+    ensure that each process uses a different subset of the dataset, providing
+    you first call ``options.randomSeed(N)`` with an integer N which is the
+    same across all hosts.
+    Args:
```

Comment:
```suggestion
    If this data loader is used in a distributed execution environment, it will
    ensure that each process uses a different subset of the dataset, providing
    you first call ``options.randomSeed(N)`` with an integer N which is the
    same across all hosts.

    Args:
```